src/main/kotlin/org/rust/lang/core/psi/impl/RustNamedElementImpl.kt

```diff
@@ -6,12 +6,14 @@ import com.intellij.psi.PsiNameIdentifierOwner
 import org.rust.lang.core.psi.RustNamedElement
 import org.rust.lang.core.psi.RustPsiFactory
 import org.rust.lang.core.psi.RustTokenElementTypes
+import org.rust.lang.core.psi.RustTokenType
 
-abstract class RustNamedElementImpl(node: ASTNode) : RustCompositeElementImpl(node),
-                                                     RustNamedElement,
-                                                     PsiNameIdentifierOwner {
+abstract class RustNamedElementImpl(
+    node: ASTNode,
+    val idTokenType: RustTokenType = RustTokenElementTypes.IDENTIFIER
+) : RustCompositeElementImpl(node), RustNamedElement, PsiNameIdentifierOwner {
 
-    override fun getNameIdentifier(): PsiElement? = findChildByType(RustTokenElementTypes.IDENTIFIER)
```

Comment:
This non final implementation is here exactly to avoid declaring a `mixin` class. But, because you name element is identifier, you have to declare mixin, so just override this method there instead of modifying the base class. You'll need something like `nameIdentifier = lifetime` I think.